internal/test/scorecard/common_utils.go

```diff
@@ -218,42 +218,87 @@ func setupCRTestResources(tr *TestResources, openShiftCertManager bool) error {
 	return nil
 }
 
+func setupTargetNamespace(tr *TestResources, name string) error {
+	client := tr.Client
+	r := tr.TestResult
+	ctx := context.Background()
+
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	ns, err := client.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create namespace %s: %s", name, err.Error())
+	}
+	r.Log += fmt.Sprintf("create namespace: %s\n", ns.Name)
+	return nil
+}
+
 func newCryostatCR(name string, namespace string, withIngress bool) *operatorv1beta1.Cryostat {
 	cr := &operatorv1beta1.Cryostat{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
 			Namespace: namespace,
 		},
-		Spec: operatorv1beta1.CryostatSpec{
-			Minimal:           false,
-			EnableCertManager: &[]bool{true}[0],
+		Spec: newCryostatSpec(),
+	}
+
+	if withIngress {
+		newIngress(name, cr.Spec)
+	}
+	return cr
+}
+
+func newClusterCryostatCR(name string, namespace string, namespaces []string, withIngress bool) *operatorv1beta1.ClusterCryostat {
+	cr := &operatorv1beta1.ClusterCryostat{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: operatorv1beta1.ClusterCryostatSpec{
+			InstallNamespace: namespace,
+			TargetNamespaces: namespaces,
+			CryostatSpec:     newCryostatSpec(),
 		},
 	}
 
 	if withIngress {
-		pathType := netv1.PathTypePrefix
-		cr.Spec.NetworkOptions = &operatorv1beta1.NetworkConfigurationList{
-			CoreConfig: &operatorv1beta1.NetworkConfiguration{
-				Annotations: map[string]string{
-					"nginx.ingress.kubernetes.io/backend-protocol": "HTTPS",
-				},
-				IngressSpec: &netv1.IngressSpec{
-					TLS: []netv1.IngressTLS{{}},
-					Rules: []netv1.IngressRule{
-						{
-							Host: "testing.cryostat",
-							IngressRuleValue: netv1.IngressRuleValue{
-								HTTP: &netv1.HTTPIngressRuleValue{
-									Paths: []netv1.HTTPIngressPath{
-										{
-											Path:     "/",
-											PathType: &pathType,
-											Backend: netv1.IngressBackend{
-												Service: &netv1.IngressServiceBackend{
-													Name: name,
-													Port: netv1.ServiceBackendPort{
-														Number: 8181,
-													},
+		newIngress(name, cr.Spec.CryostatSpec)
+	}
+	return cr
+}
+
+func newCryostatSpec() operatorv1beta1.CryostatSpec {
+	return operatorv1beta1.CryostatSpec{
+		Minimal:           false,
+		EnableCertManager: &[]bool{true}[0],
+	}
+}
+
+func newIngress(name string, cryostatSpec operatorv1beta1.CryostatSpec) {
```

Comment:
This func takes in struct `operatorv1beta1.CryostatSpec`, which is a copied spec. It should receive a struct pointer instead?